mesa/model.py

```diff
@@ -51,12 +65,27 @@ def __init__(self, *args: Any, **kwargs: Any) -> None:
         self.running = True
         self.schedule = None
         self.current_id = 0
-        self.agents: defaultdict[type, dict] = defaultdict(dict)
+        self._agents: defaultdict[type, dict] = defaultdict(dict)
+
+        # Warning flags for current experimental features. These make sure a warning is only printed once per model.
+        self.agentset_experimental_warning_given = False
+
+    @property
+    def agents(self) -> AgentSet:
+        """Provides an AgentSet of all agents in the model, combining agents from all types."""
+        all_agents = itertools.chain(
+            *[agents_by_type.keys() for agents_by_type in self._agents.values()]
```

Comment:
the syntax itertools.chain(*[]) creates a list and immediately unpacks it. It should work better with itertools.chain(()), i.e. use a generator expression instead of a list comprehension

Reply:
Great point, it's implemented in 22b99cd.